src/integrators/hyperelastic_traction_integrator.hpp

```diff
@@ -4,26 +4,72 @@
 //
 // SPDX-License-Identifier: (BSD-3-Clause)
 
+/**
+ * @file hyperelastic_traction_integrator.hpp
+ *
+ * @brief Custom MFEM integrator for nonlinear finite deformation traction loads
+ */
+
 #include "mfem.hpp"
 
 namespace serac {
 
+/**
+ * @brief Custom MFEM integrator for nonlinear finite deformation traction loads
+ */
 class HyperelasticTractionIntegrator : public mfem::NonlinearFormIntegrator {
- private:
-  mfem::VectorCoefficient&  function_;
+private:
+  /**
+   * @brief The vector coefficient for the traction load
+   */
+  mfem::VectorCoefficient& function_;
+
+  /**
+   * @brief Working matrices for the traction calculations
+   */
   mutable mfem::DenseMatrix DSh_u_, DS_u_, J0i_, F_, Finv_, FinvT_, PMatI_u_;
-  mutable mfem::Vector      shape_, nor_, fnor_, Sh_p_, Sh_u_;
 
- public:
+  /**
+   * @brief Working vectors for the traction
+   */
+  mutable mfem::Vector shape_, nor_, fnor_, Sh_p_, Sh_u_;
```

Comment:
This could also be split into separate lines, though maybe small cleanup things like that should be in a separate PR.

Mostly unrelated, but were these working vectors/matrices declared in class scope to avoid repeated constructor calls?

Reply:
Yep, we didn't want to construct and destruct these dense operators in hot loops.